backend/src/main/java/com/soen341/instagram/service/impl/AccountService.java

```diff
@@ -23,6 +24,10 @@
 
 	public void setPassword(final String newPassword)
 	{
+		if (newPassword.length() <= 0)
+		{
+			throw new PasswordBlankException();
+		}
```

Comment:
Was this not already covered by the @NotBlank?

Reply:
Because password encoder encodes even a blank string. E.g you post "", then password encode the blank string into `akslndkasl21312aksdasjd1` which isn't blank.